optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverFactory.java

```diff
@@ -26,68 +26,60 @@
 import org.optaplanner.core.api.domain.solution.PlanningSolution;
 import org.optaplanner.core.config.SolverConfigContext;
 import org.optaplanner.core.config.solver.SolverConfig;
-import org.optaplanner.core.impl.solver.EmptySolverFactory;
-import org.optaplanner.core.impl.solver.XStreamXmlSolverFactory;
+import org.optaplanner.core.config.solver.SolverConfigs;
+import org.optaplanner.core.impl.solver.DefaultSolverFactory;
 
 /**
- * Builds {@link Solver} instances.
+ * Creates {@link Solver} instances.
+ * Most applications only need one SolverFactory.
  * <p>
- * To build an instance, use {@link #createFromXmlResource(String)} or any of the other creation methods.
- * <p>
- * Supports tweaking the configuration programmatically before a {@link Solver} instance is build.
+ * To build an instance, use {@link #createFromXmlResource(String)}.
+ * To change the configuration programmatically, create a {@link SolverConfig} with {@link SolverConfigs}
+ * and use {@link #create(SolverConfig)}.
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
  */
 public abstract class SolverFactory<Solution_> {
 
     // ************************************************************************
-    // Static creation methods: XML
+    // Static creation methods: SolverConfig
     // ************************************************************************
 
     /**
-     * Uses {@link KieServices#getKieClasspathContainer()}.
-     * @param solverConfigResource never null, a classpath resource in the {@link KieContainer}
-     * as defined by {@link ClassLoader#getResource(String)}
+     * @param solverConfig never null
      * @return never null
+     * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
      */
-    public static <Solution_> SolverFactory<Solution_> createFromKieContainerXmlResource(
-            String solverConfigResource) {
-        KieContainer kieContainer = KieServices.Factory.get().getKieClasspathContainer();
-        return new XStreamXmlSolverFactory<Solution_>(new SolverConfigContext(kieContainer))
-                .configure(solverConfigResource);
+    public static <Solution_> SolverFactory<Solution_> create(SolverConfig solverConfig) {
```

Comment:
This new method "create(SolverConfig)" is important.
Originally I called it "createFromSolverConfig(SolverConfig)" to be consistent with the other create*() methods.
But I changed my mind later on, because it's inherently what happens...

Most users won't create a SolverConfig - they 'll directly create a SolverFactory from the XML.
WDYT?

Reply:
Devil's advocate: we want most users not to use SolverConfig (it's the power user option), so is it ok to have create() be short and sweet when we want them to use createFromXmlResource() by default?

Reply:
I could see that as an argument do shorten the `createFromXmlResource` method name instead. :-)